recipe/meta.yaml

```diff
@@ -1,31 +1,30 @@
 {% set name = "typer" %}
-{% set version = "0.4.1" %}
+{% set version = "0.9.0" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 5646aef0d936b2c761a10393f0384ee6b5c7fe0bb3e5cd710b17134ca1d99cff
+  sha256: 50922fd79aea2f4751a8e0408ff10d2662bd0c8bbfa84755a699f3bada2978b2
 
 build:
   skip: True  # [py<36]
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv --no-deps
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   host:
-    - python
+    - flit-core 2.3.0
     - pip
-    - flit-core >=2,<3
+    - python
+    - setuptools
     - wheel
   run:
-    - python >=3.6
+    - python
     - click >=7.1.1,<9.0.0
```

Comment:
I would leave the extras in run: (clorama, shellingham, rich)
https://github.com/tiangolo/typer/blob/0.9.0/pyproject.toml#L67
Or at least add them to run_constrained, since there is an upper bound set.